cirq-core/cirq/transformers/routing/route_circuit_cqc.py

```diff
@@ -245,17 +245,32 @@ def _get_one_and_two_qubit_ops_as_timesteps(
         The i'th entry in the nested two-qubit and single-qubit ops correspond to the two-qubit
         gates and single-qubit gates of the i'th timesteps respectively. When constructing the
         output routed circuit, single-qubit operations are inserted before two-qubit operations.
+
+        Raises:
+            ValueError: if circuit has intermediate measurement op's that act on three or more qubits and result is stored.
         """
         two_qubit_circuit = circuits.Circuit()
         single_qubit_ops: List[List[cirq.Operation]] = []
+        current_moment = 0
+
         for moment in circuit:
+            current_moment += 1
             for op in moment:
                 timestep = two_qubit_circuit.earliest_available_moment(op)
                 single_qubit_ops.extend([] for _ in range(timestep + 1 - len(single_qubit_ops)))
                 two_qubit_circuit.append(
                     circuits.Moment() for _ in range(timestep + 1 - len(two_qubit_circuit))
                 )
-                if protocols.num_qubits(op) == 2 and not protocols.is_measurement(op):
+                if protocols.num_qubits(op) > 2 and isinstance(op.gate, ops.MeasurementGate):
```

Comment:
keep `protocols.is_measurement`

Reply:
gett back error this from ./check/mypy when I use protocols.is_measurement

cirq-core/cirq/transformers/routing/route_circuit_cqc.py:265: error: Item "Gate" of "Optional[Gate]" has no attribute "key"  [union-attr]
cirq-core/cirq/transformers/routing/route_circuit_cqc.py:265: error: Item "None" of "Optional[Gate]" has no attribute "key"  [union-attr]